### PR TITLE
SLING-12072: migrate to jakarta.json

### PR DESCRIPTION
**Pull request by apelluru**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,12 +34,10 @@
...
     
     <properties>
-        <junit.version>4.13</junit.version>
+        <junit.version>4.13.2</junit.version>
         <hamcrest.version>1.3</hamcrest.version>
         <jacoco.version>0.6.2.201302030002</jacoco.version>
-        <junit-jupiter.version>5.6.0</junit-jupiter.version>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <junit-jupiter.version>5.10.0</junit-jupiter.version>
```
Comment (jsedding) [inline pom.xml:40]: We could bump it to 5.10.1 already.

File: pom.xml [hunk 2]
```diff
@@ -296,7 +313,7 @@
         <dependency>
             <groupId>org.junit.platform</groupId>
             <artifactId>junit-platform-launcher</artifactId>
-            <version>1.6.2</version>
+            <version>1.10.0</version>
```
Comment (jsedding) [inline pom.xml:316]: 1.10.1 is already available.

File: src/it/annotations-it/pom.xml [hunk 3]
```diff
@@ -29,7 +29,6 @@
 
 
   <properties>
-    <sling.java.version>8</sling.java.version>
```
Comment (jsedding) [inline src/it/annotations-it/pom.xml:32]: Do we need to drop java 8 support for technical reasons? Otherwise, I would be inclined to keep it.

**Review by jsedding — APPROVED** [review]
LGTM! I left a few nits, but I'll leave it up to you whether you want to address them.

File: pom.xml (hunk 2 above)
Comment (apelluru) [reply pom.xml:316]: sure, bumped to `1.10.1`

File: src/it/annotations-it/pom.xml (hunk 3 above)
Comment (apelluru) [reply src/it/annotations-it/pom.xml:32]: Actually its working fine, let me keep the java 8 support.

File: pom.xml (hunk 1 above)
Comment (apelluru) [reply pom.xml:40]: After bumping, following the encounter with an issue, I've identified a test failure. I'll proceed to generate a separate ticket to address this issue. Therefore, I'll maintain the current version at 5.10.0 for the time being.

```
[ERROR] Failures:
[ERROR]   OSGiAnnotationTest.injectServices:110->withMockedFrameworkUtil:352->lambda$injectServices$1:111->assertNoFailures:382 number of tests found
Expected: a value greater than <0>
     but: <0> was equal to <0>
[INFO]
```

**Comment by apelluru** [comment]
The [builds](https://ci-builds.apache.org/blue/organizations/jenkins/Sling%2Fmodules%2Fsling-org-apache-sling-junit-core/detail/PR-20/7/pipeline/61) appear to be in good shape; however, notifications are not being sent back to the pull request.

**Pull request merged**